src/qaqc/01-organize_raw_data.Rmd

```diff
@@ -720,53 +721,71 @@ generate_histograms <- function(df, df_index) {
   return(histogram)
 }
 
-# use imap to use the index in the list for the plot titles
-# result <- imap(all_data_flagged, ~generate_histograms(.x, .y))
+
 
 # making histograms for temperatures
+## pulling all the temperature dfs from the all_data list
 temperature_hist_list <- all_data_flagged[grep("Temperature", names(all_data_summary_list), ignore.case = TRUE)]
-temperature_hist_plots <- imap(temperature_hist_list, ~generate_histograms(.x, .y))
+
+## use imap to use the index in the list for the plot titles
+## (not loving imap, if there is a better way to do this please change this)
+temperature_hist_plots <- imap(temperature_hist_list, ~generate_general_histogram(.x, .y))
 temperature_hist_plots
-## getting the histogram data
+
+## How to pull the histogram data that ggplot generates to make histograms
 temperature_hist_data <- map(temperature_hist_plots, ~ ggplot_build(.x)$data[[1]]) 
 ```
 
 ```{r}
-# Working on seasonal histograms
-winter <- c(12,1,2)
-spring <- c(3,4,5)
-summer <- c(6,7,8)
-fall <- c(9,10,11)
-seasons <- data.frame(winter, spring, summer, fall)
-
-for (i in colnames(seasons)){
-  
-  title <- i
+# Working on seasonal histograms function 
+# useful for seeing seasonal differences and how each season affects the 
+# general histogram
 
-  df <- all_data_flagged[["archery-Temperature"]] %>%
-    filter(month %in% seasons[[i]]) 
+generate_seasonal_histogram <- function(df) {
   
-  if (all(is.finite(df$mean))) {
-    
-    minimum <- floor(min(df$mean, na.rm = TRUE))
-    maximum <- ceiling(max(df$mean, na.rm = TRUE))
+  winter <- c(12,1,2)
+  spring <- c(3,4,5)
```

Comment:
I wonder if these seasons is the best way to think about it? Might be better to look at more hydrology related seasons
Snowmelt: 4,5, 6
Monsoon: 7, 8
Fall Baseflow: 9, 10, 11
Winter baseflow: 12, 1, 2, 3

Reply:
I agree

Reply:
I also agree, I will add this to the histogram issue.